__tests__/app.test.js

```diff
@@ -112,40 +112,106 @@ describe("GET /api/articles/:article_id", () => {
-          });
-      });
+      return request(app)
+        .get("/api/articles/apples")
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("bad request");
+        });
+    });
   });
 });
 
 describe("GET /api/articles", () => {
-    describe("Functionality", () => {
-      test("200: should return an object containing all articles including the following properties: author, title, article_id, topic, created_at, votes, art_img_url, comment_count, automatically sorted in reverse date order", () => {
+  describe("Functionality", () => {
+    test("200: should return an object containing all articles including the following properties: author, title, article_id, topic, created_at, votes, art_img_url, comment_count, automatically sorted in reverse date order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          body.articles.forEach((article) => {
+            expect(Object.keys(article)).toMatchObject([
+              "article_id",
+              "title",
+              "topic",
+              "author",
+              "created_at",
+              "votes",
+              "article_img_url",
+              "comment_count",
+            ]);
+          });
+          expect(body.articles.length).toBe(13);
+          expect(body.articles).toBeSortedBy("created_at", {
+            descending: true,
+          });
+        });
+    });
+  });
+});
+
+describe("GET /api/articles/:article_id/comments", () => {
+  describe("Functionality", () => {
+    test("200: should return an array containing selected comments linked to article id each including the following properties: comment_id, votes, created_at, author, body, article_id", () => {
+      return request(app)
+        .get("/api/articles/5/comments")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.comments.length).toBe(2);
+          expect(body.comments).toBeSortedBy("created_at", {
+            descending: true,
+          })
+          body.comments.forEach((comment) => {
+            expect(Object.keys(comment)).toMatchObject([
+              "comment_id",
+              "body",
+              "article_id",
+              "author",
+              "votes",
+              "created_at",
+            ]);
+          });
+        });
+    });
+    test("200: should return an empty array if no comments relate to the article id selected", () => {
+      return request(app)
+        .get("/api/articles/2/comments")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.comments.length).toBe(0);
+          body.comments.forEach((comment) => {
+            expect(Object.keys(comment)).toMatchObject([
+              "comment_id",
+              "body",
+              "article_id",
+              "author",
+              "votes",
+              "created_at",
+            ]);
+          });
+        });
+    });
+  });
```

Comment:
This is an excellent test, but there is no need for the forEach as the comments array is empty so it is never entered.